### PR TITLE
improve logging for alt rfq request

### PR DESCRIPTION
**Pull request by alexkroeger**
## Description

More helpful logging for alt RFQ requests (approximating logs for the standard RFQ implementation).

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts [hunk 1]
```diff
@@ -211,13 +215,43 @@ export async function returnQuoteFromAltMMAsync<ResponseT>(
...
-        headers: { Authorization: `Bearer ${apiKey}` },
-        timeout: maxResponseTimeMs,
-    });
+    // let response: AxiosResponse<any>;
+    const response = await axiosInstance
+        .post(`${url}/quotes`, data, {
+            headers: { Authorization: `Bearer ${apiKey}` },
+            timeout: maxResponseTimeMs,
+        })
+        .then(resp => {
```
Comment (BMillman19) [inline packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:224]: Not sure if we need this `then`, can't we just call `.catch` right after `.post` ?

Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:224]: nope 👍

**Review by BMillman19 — APPROVED** [review]

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts [hunk 2]
```diff
@@ -211,13 +215,40 @@ export async function returnQuoteFromAltMMAsync<ResponseT>(
         }
     }
 
-    const response = await axiosInstance.post(`${url}/quotes`, data, {
-        headers: { Authorization: `Bearer ${apiKey}` },
-        timeout: maxResponseTimeMs,
-    });
+    // let response: AxiosResponse<any>;
```
Comment (phil-ociraptor) [inline packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:218]: nit: looks like commented code

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts [hunk 3]
```diff
@@ -12,8 +12,11 @@ import {
     AltQuoteRequestData,
     AltQuoteSide,
     AltRfqtMakerAssetOfferings,
+    LogFunction,
 } from '../types';
 
+const SUCCESS_CODE = 201;
```
Comment (phil-ociraptor) [inline packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:18]: Would be nice if our `StatusCodes` package was more robust...

**Review by phil-ociraptor — APPROVED** [review]
Looks good! Just a couple minor nit comments

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts [hunk 4]
```diff
@@ -211,13 +215,40 @@ export async function returnQuoteFromAltMMAsync<ResponseT>(
...
+        };
+    }
+    // successful handling but no quote is indicated by status = 'rejected'
     if (response.data.status === 'rejected') {
-        throw new Error('alt MM rejected quote');
+        return {
+            data: (emptyResponse as unknown) as ResponseT,
+            // hack: set the http status to 204 no content so we can more
+            // easily track when no quote is returned
+            status: 204,
```
Comment (phil-ociraptor) [inline packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:250]: A small thing - I do think the hack will be easy to trip on in the future. If possible, could we add a log statement aboe this, similar to what we have for the if case above?

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 3 above)
Comment (PirosB3) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:18]: Great idea, isn't this code in our `StatusCodes` package?

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts [hunk 5]
```diff
@@ -211,13 +215,40 @@ export async function returnQuoteFromAltMMAsync<ResponseT>(
...
-        headers: { Authorization: `Bearer ${apiKey}` },
-        timeout: maxResponseTimeMs,
-    });
+    // let response: AxiosResponse<any>;
+    const response = await axiosInstance
+        .post(`${url}/quotes`, data, {
+            headers: { Authorization: `Bearer ${apiKey}` },
+            timeout: maxResponseTimeMs,
+        })
+        .catch(err => {
```
Comment (PirosB3) [inline packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:224]: why not use `try/catch`? if you `await` a response, a failed promise will bubble up as a thrown error

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts [hunk 6]
```diff
@@ -211,13 +215,40 @@ export async function returnQuoteFromAltMMAsync<ResponseT>(
...
 
+    // tslint:disable-next-line:custom-no-magic-numbers
+    if (response.status !== SUCCESS_CODE) {
+        const rejectedRequestInfo = {
+            status: response.status,
+            message: response.data,
+        };
+        warningLogger(rejectedRequestInfo, `Alt RFQ MM did not return a status of ${SUCCESS_CODE}`);
+        return {
+            data: (emptyResponse as unknown) as ResponseT,
```
Comment (PirosB3) [inline packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:240]: Why did you define `emptyResponse` here? can't you just do `data: {}`? both here and on L247?

**Review by PirosB3 — CHANGES_REQUESTED** [review]

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 2 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:218]: 👍

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 3 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:18]: I don't think it is

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 6 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:240]: I believe this is to appease typescript

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 5 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:224]: is there any benefit to doing so?

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 6 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:242]: no type assertion on object literals

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 3 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:18]: It's not

File: packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts (hunk 4 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts:252]: good idea! added a warning log

**Review by PirosB3 — APPROVED** [review]

**Pull request merged**